pyfactorybridge/__init__.py

```diff
@@ -231,31 +312,44 @@ def upload_save_game(
         LoadSaveGame: bool = False,
         EnableAdvancedGameSettings: bool = False,
     ) -> dict[str, str]:
-        with open(save_file_path, "rb") as save_file_handle:
-            save_file_path_obj = Path(save_file_path)
-            multiparts = {
-                "saveGameFile": (
-                    # Save file basename (including .sav)
-                    save_file_path_obj.name,
-                    # Save file content
-                    save_file_handle.read(),
-                    # Content-Type
-                    "application/octet-stream",
+        """Uploads a save game file to the server. Requires the path to the save file."""
+        try:
+            with open(save_file_path, "rb") as save_file_handle:
+                save_file_path_obj = Path(save_file_path)
+                multiparts = {
+                    "saveGameFile": (
+                        # Save file basename (including .sav)
+                        save_file_path_obj.name,
+                        # Save file content
+                        save_file_handle.read(),
+                        # Content-Type
+                        "application/octet-stream",
+                    )
+                }
+                return self.__request(
+                    function="UploadSaveGame",
+                    properties={
+                        "SaveName": SaveName
+                        or save_file_path_obj.stem,  # Save file name without file extension
+                        "LoadSaveGame": LoadSaveGame,
+                        "EnableAdvancedGameSettings": EnableAdvancedGameSettings,
+                    },
+                    multiparts=multiparts,
                 )
-            }
-            return self.__request(
-                function="UploadSaveGame",
-                properties={
-                    "SaveName": SaveName
-                    or save_file_path_obj.stem,  # Save file name without file extension
-                    "LoadSaveGame": LoadSaveGame,
-                    "EnableAdvancedGameSettings": EnableAdvancedGameSettings,
-                },
-                multiparts=multiparts,
-            )
+        except (FileNotFoundError, PermissionError, OSError):
+            raise ServerError("Cannot read file path")
```

Comment:
I dislike "ServerError" here, but is fine for now I'd say

Reply:
Agreed, I'll change that